Add Script File feature

nicolassing opened the pull request:
Hello, 

I've added scripting file-base feature in addition of inline scripting.

im-denisenko commented on lib/Elastica/AbstractScript.php:
```diff
@@ -0,0 +1,28 @@
+{
+    /**
+     * @param array|null $params
+     * @param null $id
```
`@param string $id`

im-denisenko commented:
@nicolassing Looks good, thanks.
Build failed, but this is not related to PR.
Could you please add a few functional tests and run cs fixer using `make lint` before merge?

ruflin commented:
@nicolassing Any updates here? I plan to do a release today and would like to include this. Make sure merge master as quite a few changes went into master.

nicolassing commented:
@ruflin I've added a functional test. But this functional test will not work without the script file. I've updated elasticsearch configuration and created a script file (calculate-distance.groovy) in test/data/ but this script file must be in /etc/elasticsearch/scripts. I don't know how to do this with Docker and Vagrant :(

ruflin commented:
Thx, I will have a look.

ruflin commented:
@nicolassing After fighting a  little bit with ansible I got it running here: https://github.com/ruflin/Elastica/pull/914 I had to move the files from the data to the config/scripts directory as elasticsearch loads the files from there.

ruflin commented:
I merged my pull request with the changes. Thanks for the contribution.

nicolassing commented:
:+1: Someday i'd learn how to really use ansible, vagrant, docker, etc ... Someday ... :smile:

ruflin commented:
You can directly jump to docker ;-)
